crates/extensions/c8y_mapper_ext/src/config.rs

```diff
@@ -258,27 +271,69 @@ impl C8yMapperConfig {
     }
 
     pub fn default_internal_topic_filter(
-        config_dir: &Path,
         prefix: &TopicPrefix,
     ) -> Result<TopicFilter, C8yMapperConfigError> {
-        let mut topic_filter: TopicFilter = vec![
+        let topic_filter: TopicFilter = vec![
             "c8y-internal/alarms/+/+/+/+/+/a/+",
             C8yTopic::SmartRestRequest.with_prefix(prefix).as_str(),
             &C8yDeviceControlTopic::name(prefix),
         ]
         .try_into()
         .expect("topics that mapper should subscribe to");
 
+        Ok(topic_filter)
+    }
+
+    pub fn get_topics_from_operations(
+        prefix: &TopicPrefix,
+        config_dir: &Path,
+    ) -> Result<TopicFilter, C8yMapperConfigError> {
+        let mut topic_filter = TopicFilter::empty();
+
         if let Ok(operations) = Operations::try_new(
             config_dir
                 .join(SUPPORTED_OPERATIONS_DIRECTORY)
                 .join(C8Y_CLOUD),
         ) {
-            for topic in operations.topics_for_operations() {
+            // get topics from operations that contains `on command` field in their file and ignore those that come via smartREST
+            let topics = operations.topics_for_operations();
+
+            for topic in topics {
+                // if topic contains a template for the root prefix, replace that template with mapper prefix.
+                // else if topic contains a template not for the root prefix, reject that topic when appending the topic filter
+                // else if no template is present, simply add that topic to the topic filter (assuming that provided pattern is valid)
+                let topic = topic
+                    .split_inclusive("}")
+                    .flat_map(|s| match s.find("${") {
+                        None => vec![s],
+                        Some(i) => {
+                            let (prefix, template) = s.split_at(i);
+                            vec![prefix, template]
+                        }
+                    })
+                    .map(|s| {
+                        if let Some(template) = GenericCommandState::extract_path(s) {
+                            if template.eq(".topic.root_prefix") {
+                                prefix.as_str()
+                            } else {
+                                warn!("The custom topic '{topic}' is invalid and ignored.");
+                                s
+                            }
+                        } else {
+                            s
+                        }
+                    })
+                    .collect::<String>();
+
```

Comment:
The logic here can be much simpler. You just need to replace `${.bridge.topic_prefix}` by `C8yMapperConfig::c8y_prefix`.

Reply:
resolved by c1311aaf198b4b93fd96ab439ab77f816945e7e3